crl/checker/checker.go

```diff
@@ -59,11 +59,11 @@ func Diff(old, new *x509.RevocationList) (*diffResult, error) {
 		return nil, fmt.Errorf("CRLs were not issued by same issuer")
 	}
 
-	if !old.ThisUpdate.Before(new.ThisUpdate) {
+	if old.Number.Cmp(new.Number) >= 0 {
 		return nil, fmt.Errorf("old CRL does not precede new CRL")
 	}
 
-	if old.Number.Cmp(new.Number) >= 0 {
+	if new.ThisUpdate.Before(old.ThisUpdate) {
```

Comment:
optional: Consider updating the comment on line 53-56 to indicate that the ordering here is in order of importance. I can imagine a far-future reversion of this ordering due to a refactor.